mdal/mdal_data_model.cpp

```diff
@@ -332,3 +337,293 @@ size_t MDAL::Mesh::faceVerticesMaximumCount() const
+
+MDAL::DateTime MDAL::DateTime::fromStandartValue( int year, int month, int day, int hours, int minutes, double seconds )
+{
+  DateTime dateTime;
+  if ( month > 0 && day > 0 && hours >= 0 && minutes >= 0 && seconds >= 0 )
+  {
+    dateTime.mValid = true;
+    DateTimeValues values{year, month, day, hours, minutes, seconds};
+    dateTime.setWithGregorianJulianCalendarValues( values );
+  }
+
+  return dateTime;
+}
+
+std::string MDAL::DateTime::toStandartCalendarISO8601() const
+{
+  DateTimeValues value = dateTimeGregorianJulianCalendar();
+  if ( mValid )
+    return toString( value );
+  else
+    return "none";
+}
+
+double MDAL::DateTime::toJulianDay() const
+{
+  return mJulianTime / 24.0 / 3600 / 1000;
+}
+
+MDAL::DateTime &MDAL::DateTime::operator=( const MDAL::DateTime &other )
+{
+  mJulianTime = other.mJulianTime;
+  mValid = other.mValid;
+  return *this;
+}
+
+MDAL::DateTime MDAL::DateTime::operator+( const MDAL::Duration &duration ) const
+{
+  if ( !mValid )
+    return DateTime();
+  return DateTime( mJulianTime + duration.mDuration );
+}
+
+MDAL::DateTime &MDAL::DateTime::operator+=( const MDAL::Duration &duration )
+{
+  if ( !mValid )
+    return *this;
+  mJulianTime += duration.mDuration;
+  return *this;
+}
+
+MDAL::DateTime &MDAL::DateTime::operator-=( const MDAL::Duration &duration )
+{
+  if ( !mValid )
+    return *this;
+  mJulianTime -= duration.mDuration;
+  return *this;
+}
+
+MDAL::DateTime MDAL::DateTime::operator-( const MDAL::Duration &duration ) const
+{
+  if ( !mValid )
+    return DateTime();
+  return DateTime( mJulianTime - duration.mDuration );
+}
+
+bool MDAL::DateTime::operator==( const MDAL::DateTime &other ) const
+{
+  if ( !mValid && !other.mValid )
+    return true;
+
+  return ( mValid && other.mValid ) && ( mJulianTime == other.mJulianTime );
+}
+
+bool MDAL::DateTime::operator!=( const MDAL::DateTime &other ) const
+{
+  if ( !mValid && !other.mValid )
+    return true;
+
+  return !operator==( other );
+}
+
+bool MDAL::DateTime::operator<( const MDAL::DateTime &other ) const
+{
+  if ( !mValid && !other.mValid )
+    return false;
+  return ( mValid && other.mValid ) && ( mJulianTime < other.mJulianTime );
+}
+
+bool MDAL::DateTime::operator>( const MDAL::DateTime &other ) const
+{
+  if ( !mValid && !other.mValid )
+    return false;
+  return ( mValid && other.mValid ) && ( mJulianTime > other.mJulianTime );
+}
+
+bool MDAL::DateTime::operator>=( const MDAL::DateTime &other ) const
+{
+  if ( !mValid && !other.mValid )
+    return true;
+  return ( mValid && other.mValid ) && ( mJulianTime >= other.mJulianTime );
+}
+
+bool MDAL::DateTime::operator<=( const MDAL::DateTime &other ) const
+{
+  if ( !mValid && !other.mValid )
+    return true;
+  return ( mValid && other.mValid ) && ( mJulianTime <= other.mJulianTime );
+}
+
+MDAL::DateTime::DateTime( int64_t julianTime ): mJulianTime( julianTime )
+{}
+
+MDAL::DateTime::DateTimeValues MDAL::DateTime::dateTimeGregorianJulianCalendar() const
```

Comment:
in general, we would need to create some unit tests for this data time/ duration classes. I mean we need to test directly there classes alone, so we are sure the conversion between various time stuff is valid.

Reply:
I agree, tests will be an important part of the work!